GiniVision/Classes/Core/CameraViewController.swift

```diff
@@ -574,6 +518,81 @@ extension CameraViewController {
         }
     }
     
+    fileprivate func setupCamera(giniConfiguration: GiniConfiguration) {
+        self.camera = Camera(giniConfiguration: giniConfiguration) {[weak self] error in
+            if let error = error {
+                switch error {
+                case .notAuthorizedToUseDevice:
+                    addNotAuthorizedView()
+                default:
+                    if GiniConfiguration.DEBUG { cameraState = .valid; addDefaultImage() }
+                }
+                self?.failureBlock?(error)
+            }
+        }
+        self.camera?.didDetectQR = {[weak self] qrDocument in
+            if self?.detectedQRCodeDocument != qrDocument {
+                self?.detectedQRCodeDocument = qrDocument
+                self?.didPick(qrDocument)
+            }
+        }
+    }
+    
+    @objc fileprivate func captureImage(_ sender: AnyObject) {
+        guard let camera = camera else {
+            Log(message: "No camera initialized", event: .warning)
+            return
+        }
+        
+        if GiniConfiguration.DEBUG {
+            // Retrieve image from default image view to make sure image
+            // was set and therefor the correct states were checked before.
+            if let image = self.defaultImageView?.image,
+                let imageData = UIImageJPEGRepresentation(image, 0.2) {
```

Comment:
We have had problems with the compression level in the past, mostly in the Gini Pay app. And we ran experiments to determine the best value. I think it's 0.7, but ask in your team. Or check the Gini Pay repo.

Reply:
Sure, I didn't see that it only applied to the simulator.
But does it? It's actually in debug mode. Previously, and especially in Gini Pay we used to have this piece of code but only after the camera capture failed. That meant that there was no real camera on the device (or was broken, I guess). Here, I only see the check for the debug flag.

Reply:
The `defaultImageView` is only initialized when there is no camera, that is why it can be confusing here. Let me add a check here.